server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileExporters.java

```diff
@@ -213,4 +223,22 @@ private static RuleActivation toRuleActivation(org.sonar.api.rules.ActiveRule ac
     return ruleActivation;
   }
 
+  private enum ActiveRuleDtoToId implements Function<ActiveRuleDto, Integer> {
+    INSTANCE;
+
+    @Override
+    public Integer apply(@Nonnull ActiveRuleDto input) {
+      return input.getId();
+    }
+  }
+
+  private enum ActiveRuleParamDtoToActiveRuleId implements Function<ActiveRuleParamDto, Integer> {
+    INSTANCE;
+
+    @Override
+    public Integer apply(@Nonnull ActiveRuleParamDto input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The "input" parameter nullability is different in the superclass method, and that should not be changed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)